mongodb/spring-cloud-starter-stream-source-mongodb/src/main/java/org/springframework/cloud/stream/app/mongodb/source/MongodbSourceConfiguration.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.mongodb.source;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.app.trigger.TriggerConfiguration;
+import org.springframework.cloud.stream.app.trigger.TriggerPropertiesMaxMessagesDefaultUnlimited;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.integration.core.MessageSource;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.mongodb.inbound.MongoDbMessageSource;
+import org.springframework.messaging.MessageChannel;
+
+/**
+ * A starter configuration for MongoDB Source applications.
+ * Produces MongoDbMessageSource which polls collection with the query after startup according to the polling properties.
+ *
+ * @author Adam Zwickey
+ *
+ */
+@EnableBinding(Source.class)
+@EnableConfigurationProperties({MongodbSourceProperties.class, TriggerPropertiesMaxMessagesDefaultUnlimited.class})
+@Import({ TriggerConfiguration.class })
+public class MongodbSourceConfiguration {
```

Comment:
Yeah, I'd prefer the correct naming and not that one to meet the generator plugin vulnerabilities.
I mean that the name like `MongoDBSourceConfiguration` would be much better.
For plugin we need only to add this option:
```
<mongodb-source>
    <autoConfigClass>org.springframework.cloud.stream.app.mongodb.source.MongoDBSourceConfiguration.class</autoConfigClass>
</mongodb-source>
```

I can fix that on merge, if you don't mind together with minor code style polishing (e.g. proper JavaDocs format), but it really is ready for merge.
@azwickey-pivotal confirmed me in private chat that generate app for rabbit binder works well for his external MongoDB.

@ericbottard , any last objections ?